wordpress-mobile/WordPress-Android
# AppBar refresh - Site Settings, Comments, People, Plugins

#12608 needs to be merged first.

This PR updates AppBar design and adds lift on scroll functionality to the following screens (and related nested screens):

- Site Settings
- Comments
- People
- Plugins

What is lift on scroll? When you scroll content below the AppBar its elevation changes, and as a result shadow appears in light mode, and in Dark Mode color of AppBar gets lighter.

@osullivanchris I would like your opinion on changes to the filter element in Comments and People screens. I updated the look of the filtered list according to your mock, but there is a couple of functional limitations that I was not able to overcome without rewriting the whole component:
- Filter is attached to the top of the list, and will not enter unless you scroll to the top of the list (otherwise it will look weird without having any elevation or background color for filter).
- Even where there is not enough content in the list to scroll, the filter will scroll under AppBar (this is preexisting behavior).
- Scrolling filter under AppBar will not trigger elevation change, as shown on this gif:
[![Image from Gyazo](https://i.gyazo.com/89c0b52929d3ac4c8412c5c6078d8019.gif)](https://gyazo.com/89c0b52929d3ac4c8412c5c6078d8019)

[![Image from Gyazo](https://i.gyazo.com/a4eb46a213dae4c18f19964227f2e373.png)](https://gyazo.com/a4eb46a213dae4c18f19964227f2e373)

To test:

For all screens:

Make sure nothing looks out of place (including ActionMode) :)
Make sure the toolbar state is restored on configuration change.

Test 1:

- Navigate to the Site Settings screen
- Scroll the preference list and make sure the lift on scroll works (add a couple of entries and rotate the screen into landscape mode to make the list scrollable).
- Check Tags screen and make sure that lift on scroll works
- Navigate to nested preference screen by clicking on `More` under `Discussion` category.
- Make sure that the lift on scroll in nested preference screen works as expected.
- Check Blacklist and Hold for moderation screen under `Discussion` category and make sure lift on scroll there works as expected (add a couple of entries and rotate the screen into landscape mode to make the list scrollable).

Test 2:

- Navigate to Comments screen.
- Navigate to comment details (make sure you have at least one long comment).
- Make sure the lift on scroll works when you scroll the comment.
- Swipe comments sideways (they are in ViewPager) and make sure the toolbar elevation changes based on the scrolled state of each comment.

Test 3:

- Navigate to People screen.
- Scroll the content and make sure the lift on scroll works.
- Navigate to Add Person screen, rotate your device into horizontal orientation, and make sure the lift on scroll works when you scroll the form.

Test 4:

- Navigate to Plugins screen.
- Scroll the content and make sure the lift on scroll works (you might need to rotate your device).
- Open plugin list by tapping Manage or See All buttons above plugin rows.
- Scroll the content and make sure the lift on scroll works (you might need to rotate your device).

PR submission checklist:
